examples/gno.land/p/demo/simpledao/vote.gno

```diff
@@ -0,0 +1,64 @@
+package simpledao
+
+import (
+	"errors"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/gov/dao"
+)
+
+var errAlreadyVoted = errors.New("vote already cast")
+
+// votes is a simple weighted voting system
+type votes struct {
+	// tally cache to keep track of active
+	// yes / no / abstain votes
+	yays     uint64
+	nays     uint64
+	abstains uint64
+
+	voters *avl.Tree // std.Address -> dao.VoteOption
+}
+
+// newVotes creates a new weighted voting system instance
+func newVotes() *votes {
+	return &votes{
+		voters: avl.NewTree(),
+	}
+}
+
+// castVote casts a single vote in the name of the given member
+func (v *votes) castVote(member dao.Member, option dao.VoteOption) error {
+	// Check if the member voted already
+	address := member.Address.String()
+
+	_, voted := v.voters.Get(address)
+	if voted {
+		return errAlreadyVoted
+	}
+
+	// Update the tally
+	switch option {
+	case dao.YesVote:
+		v.yays += member.VotingPower
+	case dao.AbstainVote:
+		v.abstains += member.VotingPower
+	default:
+		v.nays += member.VotingPower
+	}
+
+	// Save the voting status
+	v.voters.Set(address, option)
+
+	return nil
+}
+
+// getTally returns the yay, and nay count, respectively
+func (v *votes) getTally() (uint64, uint64, uint64) {
+	return v.yays, v.nays, v.abstains
+}
+
+// getVoters returns the current voter set
+func (v *votes) getVoters() *avl.Tree {
+	return v.voters
+}
```

Comment:
no need for these methods, fields are already accessible.

Reply:
You're right, these are useless:

[c05289f](https://github.com/gnolang/gno/pull/2581/commits/c05289f18c7cd5c13ea2baf058ae9f3c8eede71d)